### DISCUSSION TITLE
docker user after upgrade to version 3.6.0 not working

### DISCUSSION THREAD

**Discussion by kimpenhaus** [coder/code-server#2199] [Q&A] [answered]
hi all,

i am running code-server inside a docker on a synology nas. the container was running flawless - but after upgrading to v3.6.0 it stopped booting properly. I think this is depending on the change mentioned in the release notes:

- The docker image's $DOCKER_USER env variable works correctly now #2057

I've updated my docker create command accordingly (at least accordingly what I understood from the [install.md](https://github.com/cdr/code-server/blob/v3.6.0/doc/install.md))

```
sudo docker create --name=code-server  \
-v "/volume1/docker/code-server/.config:/home/coder/.config" \
-v "/volume1/docker/code-server/project:/home/coder/project" \
-u "1031:100" -e DOCKER_USER=apps \
-l traefik.enable=true -l traefik.http.routers.vscode.middlewares='https_redirect@file' \
-l traefik.http.routers.vscode.rule='Host(`vscode.xxxx.net`)' \
-l traefik.http.routers.vscode.entrypoints='web, web-secure' \
--memory=2048m --restart=unless-stopped -l com.centurylinklabs.watchtower.enable=true \
codercom/code-server:latest
```

user-uid, user-god and user-name are correct:

<img width="273" alt="image" src="https://user-images.githubusercontent.com/575109/95899823-615f8100-0d91-11eb-8a56-33ebb7335e0d.png">

docker runs as `su` on the synology but the docker should be mapped to a restricted `apps` user - which was working before the upgrade. only change I did was adding `-e DOCKER_USER=apps`. that's why it is hardcoded to `1031:100` and not to `$(id -u):$(id -g)`

maybe I misunderstood what to change after the upgrade. but with the given setup the logs are showing:

`whoami: cannot find name for user ID 1031`

any help is appreciated - thanks!

m.

**Comment by code-asher** [comment, marked as answer]
Ah yeah looks like `whoami` errors for any ID that isn't already in the
Docker container.

I'm not sure what the proper fix will be yet (maybe we can explicitly
export `USER` as `coder` in the Dockerfile and avoid `whoami`) but one
workaround could be to override the entrypoint and skip all the user
renaming code. For example:

```
docker run -u 1031:100 \
  --entrypoint /bin/bash \
  codercom/code-server \
  -c 'dumb-init fixuid -q /usr/bin/code-server'
```

**Reply by nhooyr** [reply]
Filed https://github.com/cdr/code-server/issues/2213

**Reply by kimpenhaus** [reply]
works flawless - thanks mate!
